pkg/pool-weighted/test/foundry/SeedlessLBPPool.t.sol

```diff
@@ -0,0 +1,926 @@
+
+        uint256 reserveBalanceBefore = reserveToken.balanceOf(alice);
+
+        vm.prank(alice);
+        router.swapSingleTokenExactIn(
+            pool,
+            projectToken,
+            reserveToken,
+            sellAmount,
+            0,
+            type(uint256).max,
+            false,
+            bytes("")
+        );
+
+        uint256 reserveReceived = reserveToken.balanceOf(alice) - reserveBalanceBefore;
+
+        assertGt(reserveReceived, 0, "Should receive reserve tokens");
+    }
+
+    function testSwapSellProjectTokenExceedsReserves() public {
+        // Deploy pool with bidirectional swaps enabled
+        (pool, ) = _createLBPoolWithCustomWeights(
+            address(0),
+            startWeights[projectIdx],
+            startWeights[reserveIdx],
+            endWeights[projectIdx],
+            endWeights[reserveIdx],
+            uint32(block.timestamp + DEFAULT_START_OFFSET),
+            uint32(block.timestamp + DEFAULT_END_OFFSET),
+            false // Allow project token swaps in
+        );
+        initPool();
+
+        // Warp to when swaps are enabled
+        vm.warp(block.timestamp + DEFAULT_START_OFFSET);
+
+        uint256 buyAmount = 1e18;
+        uint256 sellAmount = 10e18; // Should fail with sellAmount >> buyAmount
+
+        // First, user buys project tokens to build real reserve
+        vm.prank(bob);
+        router.swapSingleTokenExactIn(
+            pool,
+            reserveToken,
+            projectToken,
+            buyAmount,
+            0,
+            type(uint256).max,
+            false,
+            bytes("")
+        );
+
+        uint256 snapshotId = vm.snapshotState();
+
+        uint256[] memory balances = vault.getCurrentLiveBalances(pool);
+
+        uint256 effectiveReserveBalance = balances[reserveIdx] + reserveTokenVirtualBalance;
+
+        uint256 expectedAmountOut = WeightedMath.computeOutGivenExactIn(
+            balances[projectIdx],
+            startWeights[projectIdx],
+            effectiveReserveBalance,
+            startWeights[reserveIdx],
+            sellAmount.mulDown(DEFAULT_SWAP_FEE_PERCENTAGE.complement())
+        );
+
+        vm.revertToState(snapshotId);
+
+        vm.expectRevert(
+            abi.encodeWithSelector(
+                ILBPool.InsufficientRealReserveBalance.selector,
+                expectedAmountOut,
+                balances[reserveIdx]
+            )
+        );
+        vm.prank(alice);
+        router.swapSingleTokenExactIn(
+            pool,
+            projectToken,
+            reserveToken,
+            sellAmount,
+            0,
+            type(uint256).max,
+            false,
+            bytes("")
+        );
+    }
+
+    /*******************************************************************************
+                                    Price Verification
+    *******************************************************************************/
+
+    function testPriceUsesVirtualBalance() public {
```

Comment:
👍  this is fine.

I'd add one more test: create a seedless LBP and a mirror LBP with real reserves and 0 virtual reserves.
At any given point in time during the sell, swaps exact in and out should return the exact same result.

This indirectly tests that virtual reserves are indistinguishable from real reserves when it comes to pricing. And this can be fuzzed: amounts in / out, sale timestamp, token decimals, etc.